iosApp/iosApp/Pages/StopDetails/StopDetailsPage.swift

```diff
@@ -74,28 +74,44 @@ struct StopDetailsPage: View {
             )
             .onAppear {
                 loadGlobalData()
-                changeStop(stop)
+                fetchStopData(stop)
                 loadPinnedRoutes()
                 didAppear?(self)
             }
-            .onChange(of: stop) { nextStop in changeStop(nextStop) }
-            .onChange(of: globalResponse) { _ in updateDepartures() }
-            .onChange(of: pinnedRoutes) { _ in updateDepartures() }
+            .onChange(of: stop) { nextStop in
+                changeStop(nextStop)
+            }
+            .onChange(of: globalResponse) { _ in
+
+                updateDepartures(stop)
+            }
+            .onChange(of: pinnedRoutes) { _ in
+                updateDepartures(stop)
+            }
             .onChange(of: predictionsByStop) { newPredictionsByStop in
                 updateDepartures(stop, newPredictionsByStop, predictions)
             }
-            .onChange(of: predictions) { _ in updateDepartures() }
-            .onChange(of: schedulesResponse) { _ in updateDepartures() }
+            .onChange(of: predictions) { _ in
+
+                updateDepartures(stop)
+            }
+            .onChange(of: schedulesResponse) { _ in
+
+                updateDepartures(stop)
+            }
             .onReceive(inspection.notice) { inspection.visit(self, $0) }
-            .task {
+            .task(id: stop.id) {
```

Comment:
This was surprising to me: without passing `stop.id` here, the contained call to `updateDepartures()` was always using the original stop from when the view was created, resulting in the old stop pills flashing in.